### PR TITLE
OPSEXP-2785 Add customizable probes for charts

### PR DESCRIPTION
**Pull request by pmacius**
ref: OPSEXP-2785

### CODE REVIEW

**Comment by pmacius** [comment]
Not sure if to touch the aps charts

**Comment by alxgomz** [comment]
> Not sure if to touch the aps charts

The recent work done probes for APS is enough for me. Not sure why someone would want to use a completely different probe for these

File: charts/alfresco-ai-transformer/templates/deployment-ai-transformer.yaml [hunk 1]
```diff
@@ -87,18 +87,18 @@ spec:
             {{- toYaml .Values.extraVolumeMounts | nindent 12 }}
           readinessProbe:
             httpGet:
-              path: /ready
+              path: {{ .Values.readinessProbe.path }}
               port: {{ .Values.image.internalPort }}
             initialDelaySeconds: {{ .Values.readinessProbe.initialDelaySeconds }}
             periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
             timeoutSeconds: {{ .Values.readinessProbe.timeoutSeconds }}
```
Comment (gionn) [inline charts/alfresco-ai-transformer/templates/deployment-ai-transformer.yaml:94]: add failureThreshold

File: charts/alfresco-search-enterprise/Chart.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ apiVersion: v2
 name: alfresco-search-enterprise
 description: A Helm chart for deploying Alfresco Elasticsearch connector
 type: application
-version: 4.1.0
+version: 5.0.0-alpha.0
```
Comment (gionn) [inline charts/alfresco-search-enterprise/Chart.yaml:6]: technically not a breaking change in this case

File: charts/alfresco-search-service/templates/deployment.yaml [hunk 3]
```diff
@@ -89,9 +89,7 @@ spec:
                   | xmllint --xpath '/response/str[@name="status"]/text()' -
                   )
                   = "OK" ]
```
Comment (gionn) [inline charts/alfresco-search-service/templates/deployment.yaml:91]: not sure what to do either, if you move them you also need the tpl filter

File: charts/activemq/templates/deployment-activemq.yaml [hunk 4]
```diff
@@ -55,20 +55,10 @@ spec:
...
-          tcpSocket:
-            port: {{ .Values.services.broker.ports.internal.openwire | default 61616 }}
-          initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
-          periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
-          failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
-          timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+          {{- toYaml .Values.livenessProbe | nindent 12 }}
+        readinessProbe:
+          {{- toYaml .Values.readinessProbe | nindent 12 }}
         resources: {{- toYaml .Values.resources | nindent 12 }}
```
Comment (gionn) [inline charts/activemq/templates/deployment-activemq.yaml:62]:
```suggestion
        resources:
          {{- toYaml .Values.resources | nindent 12 }}
```

🥶

File: charts/alfresco-adf-app/templates/deployment.yaml [hunk 5]
```diff
@@ -42,12 +42,20 @@ spec:
               protocol: TCP
           livenessProbe:
             httpGet:
-              path: /
+              path: {{ .Values.livenessProbe.path }}
               port: {{ .Values.service.targetPort }}
+            initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+            failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+            timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
```
Comment (gionn) [inline charts/alfresco-adf-app/templates/deployment.yaml:50]: why not using the toYaml syntax also here? you can use the named port as well

File: charts/alfresco-ai-transformer/templates/deployment-ai-transformer.yaml [hunk 6]
```diff
@@ -87,18 +87,19 @@ spec:
             {{- toYaml .Values.extraVolumeMounts | nindent 12 }}
           readinessProbe:
             httpGet:
-              path: /ready
+              path: {{ .Values.readinessProbe.path }}
               port: {{ .Values.image.internalPort }}
```
Comment (gionn) [inline charts/alfresco-ai-transformer/templates/deployment-ai-transformer.yaml:91]: use the named port & toyaml

File: charts/alfresco-connector-ms365/templates/deployment-connector-ms365.yaml [hunk 7]
```diff
@@ -34,7 +34,8 @@ spec:
           env:
             {{- include "alfresco-connector-ms365.env" $ | indent 12 }}
           ports:
-              - containerPort: {{ .Values.image.internalPort }}
+            - name: {{ .Values.readinessProbe.tcpSocket.port }}
```
Comment (gionn) [inline charts/alfresco-connector-ms365/templates/deployment-connector-ms365.yaml:37]: would be a bit confusing to use this here, I would hardcode the nameport and have such value in Values.readinessProbe.tcpSocket.port

**Review by gionn — CHANGES_REQUESTED** [review]
try to be consistent with the approach, I would avoid the toYaml only if at least path is not hardcoded and is using the named port

File: charts/alfresco-adf-app/templates/deployment.yaml (hunk 5 above)
Comment (pmacius) [reply charts/alfresco-adf-app/templates/deployment.yaml]: Using the named port here breaks the probes, it has to connect to port 8080

File: charts/alfresco-adf-app/values.yaml [hunk 8]
```diff
@@ -46,6 +46,24 @@ ingress:
           pathType: Prefix
   tls: []
 
+livenessProbe:
+  httpGet:
+    path: /
+    port: 8080
```
Comment (gionn) [inline charts/alfresco-adf-app/values.yaml:52]:
```suggestion
    port: http
```

File: charts/alfresco-adf-app/values.yaml [hunk 9]
```diff
@@ -46,6 +46,24 @@ ingress:
...
+  initialDelaySeconds: 0
+  periodSeconds: 10
+  timeoutSeconds: 1
+  failureThreshold: 3
+
+readinessProbe:
+  httpGet:
+    path: /
+    port: 8080
```
Comment (gionn) [inline charts/alfresco-adf-app/values.yaml:61]:
```suggestion
    port: http
```

File: charts/alfresco-ai-transformer/Chart.yaml [hunk 10]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: alfresco-ai-transformer
 description: A Helm chart for deploying Alfresco ai transformer service
 type: application
-version: 2.0.1
+version: 2.1.0-alpha.0
```
Comment (gionn) [inline charts/alfresco-ai-transformer/Chart.yaml:5]: major?

File: charts/alfresco-share/Chart.yaml [hunk 11]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: alfresco-share
 description: Alfresco Share Helm chart for Kubernetes
 type: application
-version: 1.1.1
+version: 2.0.0-alpha.0
```
Comment (gionn) [inline charts/alfresco-share/Chart.yaml:5]: is it a major? you just added new values without removing any

File: charts/alfresco-sync-service/Chart.yaml [hunk 12]
```diff
@@ -8,7 +8,7 @@ keywords:
 name: alfresco-sync-service
 sources:
   - https://github.com/Alfresco/acs-deployment
-version: 6.0.1
+version: 7.0.0-alpha.0
```
Comment (gionn) [inline charts/alfresco-sync-service/Chart.yaml:11]: major should not be required in this case

File: charts/activemq/Chart.yaml [hunk 13]
```diff
@@ -7,7 +7,7 @@ keywords:
 name: activemq
 sources:
   - https://github.com/Alfresco/alfresco-helm-charts
-version: 3.5.5
+version: 4.0.0-alpha.0
```
Comment (gionn) [inline charts/activemq/Chart.yaml:10]: there is no breaking change in the values here

File: charts/alfresco-adf-app/templates/deployment.yaml [hunk 14]
```diff
@@ -38,16 +38,12 @@ spec:
           {{- end }}
           ports:
             - name: http
-              containerPort: {{ .Values.service.port }}
+              containerPort: {{ .Values.service.targetPort }}
```
Comment (gionn) [inline charts/alfresco-adf-app/templates/deployment.yaml:41]: I am confused, this should be the same port as the port exposed by the image itself - which should be 80 - how the application can be reachable if this one is now 8080?

**Review by gionn — APPROVED** [review]

**Pull request merged**